examples/jax/run_maddpg_test.py

```diff
@@ -0,0 +1,72 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(Arnu): remove mypy ignore when the system is running
+# type: ignore
+
+"""Test for running Jax MADDPG on debug MPE environment."""
+from datetime import datetime
+
+import optax
+
+import mava.utils as utils
+from mava.systems.tf import maddpg
+
+
+def test_main() -> None:
+    """Acceptance test for Jax MADDPG system."""
+
+    # Create system and config
+    system = maddpg.MADDPG()
+
+    # Environment.
+    environment_factory = utils.make_factory(
+        utils.environments.make_environment,
+        env_name="debugging",
+        action_space="continuous",
+    )
+
+    # Logger.
+    logger_factory = utils.make_factory(
+        utils.loggers.make_logger,
+        directory="~/mava",
+        to_terminal=True,
+        to_tensorboard=True,
+        time_stamp=str(datetime.now()),
+        time_delta=10,
+    )
+
+    # Networks.
+    network_factory = utils.make_factory(maddpg.make_default_networks)
+
+    # optimizers
+    policy_optimizer = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
+    critic_optimizer = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
```

Comment:
I'm a bit confused by this: are we using `tf` networks but `jax` optimisers?

Reply:
No, in this case, the maddpg is a Jax version and its `make_default_networks` will be jax based, probably using haiku.

Reply:
I have the same question. In this example (and when I pulled the branch), it is linking to our TF MADDPG. Test therefore fails as the factories are not included in the MADDPG args and it cannot find the system (jax) methods. Only one file is changed as well?

Reply:
My bad, I saw now I had a confusing `from mava.systems.tf import maddpg` which wasn't suppose to be importing from `tf` but from `jax`. Fixed that now. @AsadJeewa, regarding dependencies etc. in your other comments, note this acceptance test is expected to fail for quite some time as there are no Jax system implementations yet. Once we start with those we will start updating dependencies to support optax etc.

Reply:
The comment got deleted by mistake so adding back for reference. "I had the same question. I see that we are implementing tf MADDPG and the method subsequently fails since the factories are not included in the MADDPG method. I also see only one file changed so maybe I am missing something"

Reply:
It's a new way of thinking since it feels weird pushing code with files that do not yet exist but I get that this is what TDD requires. So we would we right the failing test without Jax MADDPG implemented and next step, we write the class to pass the test, then iteratively improve?